pilot/pkg/model/push_context.go

```diff
@@ -754,7 +754,10 @@ func virtualServiceDestinations(v *networking.VirtualService) map[string]sets.Se
 		return nil
 	}
 
-	out := make(map[string]sets.Set[int])
+	// In most cases, there will be only one `Route`. Weight-based routing may have multiple `Route`,
+	// but 2 is more common. This estimation method can avoid capacity growth in most scenarios.
+	estimatedCap := len(v.Http) + len(v.Tcp) + len(v.Tls) + 1
```

Comment:
In our production environment, some services were migrated from nginx-based gateway to istio. Since the original gateway had many http path-based routes, we observed that there would occasionally be very slight performance issues here.